From 4843ba67971ed6fc2f69ee283862e70053270c0d Mon Sep 17 00:00:00 2001
From: savlpavel <savlpavel@gmail.com>
Date: Fri, 25 Nov 2016 16:02:36 +0300
Subject: [PATCH] add "Choose Db" button

---
 Forms/Profile Document List.form | 110 +++++++++++++++++++++++++--------------
 1 file changed, 70 insertions(+), 40 deletions(-)

diff --git a/Forms/Profile Document List.form b/Forms/Profile Document List.form
index f3505ba..6810f3e 100644
--- a/Forms/Profile Document List.form	
+++ b/Forms/Profile Document List.form	
@@ -1,22 +1,23 @@
 <?xml version='1.0' encoding='utf-8'?>
 <form name='Profile Document List' alias='ProfileDocManager' xmlns='http://www.lotus.com/dxl'
  version='9.0' maintenanceversion='1.7' replicaid='432580060036840F' noquery='true'
- otherdialog='true' publicaccess='false' designerversion='8' bgcolor='#e0f1ff'
+ otherdialog='true' publicaccess='false' designerversion='8.5.3' bgcolor='#e0f1ff'
  renderpassthrough='true'>
-<noteinfo noteid='1196' unid='8AD2F0DBF45CA93F43258076003A8AD9' sequence='2'>
+<noteinfo noteid='1196' unid='8AD2F0DBF45CA93F43258076003A8AD9' sequence='34'>
 <created><datetime>20161125T133926,33+03</datetime></created>
-<modified><datetime>20161125T134040,27+03</datetime></modified>
-<revised><datetime>20161125T134040,26+03</datetime></revised>
-<lastaccessed><datetime>20161125T134040,27+03</datetime></lastaccessed>
-<addedtofile><datetime>20161125T133926,66+03</datetime></addedtofile></noteinfo>
-<updatedby><name>CN=Andre Guirard/OU=Cambridge/O=IBM</name><name>CN=Pavel Aleshkevich/O=tst/C=RU</name></updatedby>
+<modified><datetime>20161125T160109,02+03</datetime></modified>
+<revised><datetime>20161125T160109,01+03</datetime></revised>
+<lastaccessed><datetime>20161125T160109,02+03</datetime></lastaccessed>
+<addedtofile><datetime>20161125T154626,45+03</datetime></addedtofile></noteinfo>
+<updatedby><name>CN=Pavel Aleshkevich/O=tst/C=RU</name></updatedby>
 <wassignedby><name>CN=Pavel Aleshkevich/O=tst/C=RU</name></wassignedby>
 <globals><code event='options'><lotusscript>Option Public
 Option Declare
 %INCLUDE "lsconst"
 </lotusscript></code><code event='declarations'><lotusscript>Dim Gchoices ' choices as they would appear in the selection box, with noteID as keyword synonym
 Dim GdbTarget As NotesDatabase
-
+Dim currDoc As NotesDocument
+Dim currUIDoc As NotesUIDocument
 Const WINDOWTITLE = "Profile Manager"
 
 Private Const TIME_CLOCK_START = 4. / (24. * 60. * 60.) ' 4 seconds (or 1%, whichever is more)
@@ -301,12 +302,37 @@ End Sub
 	' as choices in the Selection field.
 	Source.Document.ReplaceItemValue "Choices", Gchoices
 	Source.Refresh
-End Sub</lotusscript></code>
+	Set currUIDoc = Source
+	Set currDoc = currUIDoc.Document
+End Sub
+</lotusscript></code>
 <actionbar bgcolor='silver' bordercolor='black'>
 <actionbuttonstyle bgcolor='silver'/><font color='system'/><border style='solid'
- width='0px 0px 1px'/>
+ width='0px 0px 1px' color='system'/>
 <action title='Select All' icon='10'><code event='click'><formula>FIELD Selection := @Right(Choices; "|");
 @Command([RefreshHideFormulas])</formula></code></action>
+<action title='Choose Db'><code event='options'><lotusscript>Option Declare
+
+</lotusscript></code><code event='click'><lotusscript>Sub Click(Source As Button)
+	Dim ws As New NotesUIWorkspace
+	Dim ses As New NotesSession
+	
+	Dim res As Variant
+	
+	res = ws.Prompt(13,"","")
+	If Isempty(res) Then Exit Sub
+	
+	Set GdbTarget = ses.GetDatabase(res(0), res(1), False)
+	If GdbTarget Is Nothing Then Exit Sub
+	If Not GdbTarget.IsOpen() Then Exit Sub
+	
+	If calculateChoices(GdbTarget) = 0 Then
+		Msgbox "There are no profile documents in '" &amp; GdbTarget.Title &amp; {.}, 0, WINDOWTITLE
+		Exit Sub
+	End If
+	
+	Call currUIDoc.Refresh
+End Sub</lotusscript></code></action>
 <action title='Clear' icon='11'><code event='click'><formula>FIELD Selection := "";
 @Command([RefreshHideFormulas])</formula></code></action>
 <action title='Edit Profile' icon='89'><code event='options'><lotusscript
@@ -381,7 +407,8 @@ End Sub</lotusscript></code>
 			End If
 		End If
 	End If
-End Sub</lotusscript></code><code event='hidewhen'><formula>@Elements(Selection) != 1</formula></code></action>
+End Sub
+</lotusscript></code><code event='hidewhen'><formula>@Elements(Selection) != 1</formula></code></action>
 <action title='Delete Selected' icon='116'><code event='options'><lotusscript
 >Option Declare
 
@@ -412,7 +439,8 @@ End Sub</lotusscript></code><code event='hidewhen'><formula>@Elements(Selection)
 		End Forall
 		RefreshChoices uidoc
 	End If
-End Sub</lotusscript></code><code event='hidewhen'><formula>!(Selection != "")</formula></code></action>
+End Sub
+</lotusscript></code><code event='hidewhen'><formula>!(Selection != "")</formula></code></action>
 <action title='Refresh' icon='145'><code event='options'><lotusscript>Option Declare
 
 </lotusscript></code><code event='click'><lotusscript>Sub Click(Source As Button)
@@ -432,7 +460,8 @@ End Sub</lotusscript></code><code event='hidewhen'><formula>!(Selection != "")</
 	Else
 		Msgbox "No more profile documents!", MB_ICONINFORMATION, WINDOWTITLE
 	End If
-End Sub</lotusscript></code></action></actionbar>
+End Sub
+</lotusscript></code></action></actionbar>
 <body><richtext>
 <pardef id='1'/>
 <par def='1'><run><font size='14pt' color='maroon'/>Profile Document Management</run></par>
@@ -488,7 +517,8 @@ FIELD Choices := _sorted;
 '	Dim uidoc As NotesUIDocument
 '	Set uidoc = wksp.CurrentDocument
 '	uidoc.RefreshHideFormulas
-End Sub</lotusscript></code></field></run></par>
+End Sub
+</lotusscript></code></field></run></par>
 <par def='2'><run><font size='8pt' color='red'/>hidden: </run><run><font size='8pt'
  color='red'/><field type='text' kind='computedwhencomposed' name='SaveOptions'><code
  event='defaultvalue'><formula>"0"</formula></code></field></run><compositedata
@@ -497,36 +527,36 @@ End Sub</lotusscript></code></field></run></par>
 Yg4BAIQAAAAAAAAAAAA=
 </compositedata><run><font size='8pt'
  color='red'/></run></par></richtext></body>
+<item name='$$ScriptName' summary='false' sign='true'><text>ProfileDocManager</text></item>
 <item name='$Selection_O' sign='true'>
 <rawitemdata type='14'>
-AQAAAkxTT0IDABQIZW4AAAQACQAoAAQAAAAkAowBAAAAAAQAwADMAAAAAAAoAAAAHAA8AMgANAFY
-AAAAAAAAAAIAAAAYAGwBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
-AAAAAAAAGABsAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
-AAAAAAAAAAAAAAAAAAEAAADMAMwAAAAAAMwAzAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+AQAAAkxTT0IDABQIZW4AAAQACQAoAAQAAAAkAogBAAAAAAQAGAAkAAAAAAAkADwBGAA4AMQAbAEA
+AAAAAAAAAAIAAAC0AFwBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+AAAAAAAAtABcAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+AAAAAAAAAAAAAAAAAAEAAAAkACQAAAAAACQAJAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
 AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
 AAAAAAAAAAAAAAAAAAAAAAAAAAEABQBMb3R1cyBQcm9kdWN0AAAAVjAuMAAAAAAAAAAAAAAAAAAA
-AAAEAAAAKAAAAAAAAAAkAgAAjAEAAAAAAAAAAAAAAQAAAAAAAAABAAAAAAAAAAEAAAAAAAAAAAAA
+AAAEAAAAKAAAAAAAAAAkAgAAiAEAAAAAAAAAAAAAAQAAAAAAAAABAAAAAAAAAAEAAAAAAAAAAAAA
 AAUAAABXMzJJJQMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
 AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
-AAAAAwCQAQAAAAAYAQsAJAAkAFMARQBMAEUAQwBUAEkATwBOAAAAcAADAE4ARQBXAAAAhAAGAEQA
-RQBMAEUAVABFAAAAAAD//woASQBOAEkAVABJAEEATABJAFoARQAAAAAA//8JAFQARQBSAE0ASQBO
-AEEAVABFAAAAjAAGAE8AQgBKAEUAQwBUAAAAAAAEAQAAAAAAAP//GgAoAEcATABPAEIAQQBMAFMA
-KQBQAFIATwBGAEkATABFAEQATwBDAE0AQQBOAEEARwBFAFIAAAAAAHQBGgAoAEcAbABvAGIAYQBs
-AHMAKQBQAHIAbwBmAGkAbABlAEQAbwBjAE0AYQBuAGEAZwBlAHIAAAAAAFABBgBTAE8AVQBSAEMA
-RQAAAAAAYAEKAEIASQBOAEQARQBWAEUATgBUAFMAAAAAAP//CwBPAEIASgBFAEMAVABOAEEATQBF
-AF8AAAD//wUARgBJAEUATABEAAAA//8GACUATABTAEkARABFAAAAAAD//wgATwBOAEMASABBAE4A
-RwBFAAAAAAAFACgCAAAAAGik3XAYAAAAAACQAIiCxgg4x2ICBAAAAAgAAABsAWwBHAEAAAAAAAAC
-AAAAsABcAQAAAAAAAAAAAAAAAAAAAACwALAAAAAAAAAAAAAAAAAAAAAAAAAAAABcAVwBAAAAAAAA
-AAABAAAAAgAAAAYAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAwABAFwBAAA4AQYIAAAAABoAAABkAf//AAAAABAAAAAAAAAA
-VAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
-AAAAAAAAAAAAAAADFhMp7S5pEL9dAN0BEYa3AAAAAAAAAAAAAAAAAAAAAGQAAAAAAAAAAAAAAAAA
-AAACAMAAAAAUAgAAAABcBAEA//8AAAMAIAAAAAAACAEJAgAAzAAIAAAAAAAAAHgBAAAAAAAAAQAA
-AAQCBAIAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAIEAgAAAAAAAAAA
-AQAAAAAAAAAkAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
-AAAAAAAAAAAAAAAAAAAAAAAAAAMAAQAAAAAACAEJAgAAzAATAAAAeAEAAOwAAAAAAAAABAAsAAAA
-AADSzAAdAAAaDABbXAFJsADKzACmGg0AR1wBy+wAAABsARoOAB0aFgAdAgAAABEAGAAAAAAA//8B
-AAMABAAAAAcAHAEJAHgBEAAKAAAA
-</rawitemdata></item>
-<item name='$$ScriptName' summary='false' sign='true'><text>ProfileDocManager</text></item></form>
+AAAAAwCMAQAAAABUAAgAKgA4ADYAQQBGADAAMQA4AAAAAABsAAMATgBFAFcAAACAAAYARABFAEwA
+RQBUAEUAAAAAAP//CgBJAE4ASQBUAEkAQQBMAEkAWgBFAAAAAAD//wkAVABFAFIATQBJAE4AQQBU
+AEUAAACIAAYATwBCAEoARQBDAFQAAAAAAAABAAAAAAAA//8aACgARwBMAE8AQgBBAEwAUwApAFAA
+UgBPAEYASQBMAEUARABPAEMATQBBAE4AQQBHAEUAUgAAAAAAFAEaACgARwBsAG8AYgBhAGwAcwAp
+AFAAcgBvAGYAaQBsAGUARABvAGMATQBhAG4AYQBnAGUAcgAAAAAALAEGAFMATwBVAFIAQwBFAAAA
+AAD//wgATwBOAEMASABBAE4ARwBFAAAAAAD//wUARgBJAEUATABEAAAAUAEGACUATABTAEkARABF
+AAAAAAD//woAQgBJAE4ARABFAFYARQBOAFQAUwAAAAAA//8LAE8AQgBKAEUAQwBUAE4AQQBNAEUA
+XwAAAAUAKAIAAAAAaCzeTRgAAAAAAIwA8PPnFQAAAAAAAAAAGgAAAEAB//8AAAAAEAAAAAAAAAAw
+AQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+AAAAAAAAAAAAAAMWEyntLmkQv10A3QERhrcAAAAAAAAAAAAAAAAAAAAAZAAAAPCigmEAAAAAAAAA
+AAIAGAAAABQCAAAAAFwEAQD//wAACAAAAFwBXAEYAQAAAAAAAAEAAABMAUwBAAAAAAAAAAAAAAAA
+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEwBTAEAAAAAAAAAAAEAAAAAAAAABgAAAAAAAAAA
+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+AAADAAEAAAAAAAQBCQIAACQACAAAAAAAAABUAQAAAAAAAAIAAAD0AQQCAAAAAAAAAAAAAAAAAAAA
+APQB9AEAAAAAAAAAAAAAAAAAAAAAAAAAAAQCBAIAAAAAAAAAAAEAAAACAAAACgAAAAAAAAAAAAAA
+AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD
+AAEABAIAAHABBggAAAAAAwAgAAAAAAAEAQkCAAAkABMAAAAYAQAA7AAAAAAAAAAEACwAAAAAANLI
+AB0AABoPAB0aEwBbBAJJ9AHKJACmGhQARwQCy+wAAAC0ABoVAB0CAAAAEQAYAAAAAAD//wEAAwAE
+AAAABwAYAQkAVAEQAAoAAAA=
+</rawitemdata></item></form>
 
-- 
2.8.1.windows.1

